### PR TITLE
enh(api): create interfaces for update OpenId Configuration use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to create interfaces for update OpenId Configuration use case.

**Fixes** # MON-12485

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: config/routes/Centreon/administration/provider_configuration.yaml [hunk 1]
```diff
@@ -8,4 +8,10 @@ centreon_security_update_provider_configuration_local:
     methods: PUT
     path: /administration/authentication/providers/local
     controller: 'Core\Infrastructure\Security\ProviderConfiguration\Local\Api\UpdateConfiguration\UpdateConfigurationController'
+    condition: "request.attributes.get('version') >= 22.04"
+
+centreon_security_update_provider_configuration_openid:
+    methods: PUT
+    path: /administration/authentication/providers/openid
+    controller: 'Core\Infrastructure\Security\ProviderConfiguration\OpenId\Api\UpdateConfiguration\UpdateOpenIdConfigurationController'
```
Comment (kduret) [inline config/routes/Centreon/administration/provider_configuration.yaml:16]:
```suggestion
    controller: 'Core\Infrastructure\Security\ProviderConfiguration\OpenId\Api\UpdateConfiguration\UpdateConfigurationController'
```

File: src/Centreon/Application/Controller/AbstractController.php [hunk 2]
```diff
@@ -76,4 +79,38 @@ protected function getBaseUri(): string
 
         return $baseUri;
     }
+
+    /**
+     * Validate the data sent.
+     *
+     * @param Request $request Request sent by client
+     * @param string $jsonValidationFile Json validation file
+     * @throws \Exception
```
Comment (kduret) [inline src/Centreon/Application/Controller/AbstractController.php:88]:
```suggestion
     * @throws \InvalidArgumentException
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+namespace Core\Domain\Security\ProviderConfiguration\OpenId\Model;
+
+class OpenIdConfiguration
+{
+    public const DEFAULT_LOGIN_GLAIM = 'preferred_username';
+    /**
+     * @param boolean $isActive
+     * @param boolean $isForced
+     * @param array<string> $trustedClientAddresses
+     * @param array<string> $blacklistClientAddresses
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:34]:
```suggestion
     * @param string[] $trustedClientAddresses
     * @param string[] $blacklistClientAddresses
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+     * @param string|null $baseUrl
+     * @param string|null $authorizationEndpoint
+     * @param string|null $tokenEndpoint
+     * @param string|null $introspectionTokenEndpoint
+     * @param string|null $userInformationsEndpoint
+     * @param string|null $endSessionEndpoint
+     * @param array<string> $connectionScope
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:41]:
```suggestion
     * @param string[] $connectionScope
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+     * @return boolean
+     */
+    public function isForced(): bool
+    {
+        return $this->isForced;
+    }
+
+    /**
+     * @return array<string>
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:86]:
```suggestion
     * @return string[]
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+     */
+    public function getTrustedClientAddresses(): array
+    {
+        return $this->trustedClientAddresses;
+    }
+
+    /**
+     * @return array<string>
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:94]:
```suggestion
     * @return string[]
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+     * @return string|null
+     */
+    public function getEndSessionEndpoint(): ?string
+    {
+        return $this->endSessionEndpoint;
+    }
+
+    /**
+     * @return array<string>
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:150]:
```suggestion
     * @return string[]
```

File: src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        private bool $isForced,
+        private array $trustedClientAddresses,
+        private array $blacklistClientAddresses,
+        private ?string $baseUrl,
+        private ?string $authorizationEndpoint,
+        private ?string $tokenEndpoint,
+        private ?string $introspectionTokenEndpoint,
+        private ?string $userInformationsEndpoint,
+        private ?string $endSessionEndpoint,
+        private array $connectionScope,
```
Comment (kduret) [inline src/Core/Domain/Security/ProviderConfiguration/OpenId/Model/OpenIdConfiguration.php:59]: maybe renamed to `$connectionScopes`

File: src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationSchema.json [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        "is_forced",
+        "trusted_client_addresses",
+        "blacklist_client_addresses",
+        "base_url",
+        "authorization_endpoint",
+        "token_endpoint",
+        "introspection_token_endpoint",
+        "userinfo_endpoint",
+        "endsession_endpoint",
+        "connection_scope",
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationSchema.json:17]: `connection_scopes` ?

File: tests/php/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationControllerTest.php [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    UpdateOpenIdConfigurationController
+};
+use Core\Application\Security\ProviderConfiguration\OpenId\UseCase\UpdateOpenIdConfiguration\{
+    UpdateOpenIdConfiguration,
+    UpdateOpenIdConfigurationPresenterInterface
+};
+
+class UpdateOpenIdConfigurationControllerTest extends TestCase
+{
+
```
Comment (kduret) [inline tests/php/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationControllerTest.php:43]:
```suggestion
```

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**